### PR TITLE
Fixing some reported issues

### PR DESCRIPTION
**Pull request by JTPetter**
- Fixes https://github.com/jasp-stats/jasp-issues/issues/2273: Fixed calculation of y-axis limits of variance component graph in gauge r&R. It sometimes did not include zero in the limits and then removed all bars.
- Updated unit tests as graphs changed slightly after using jasp_theme_raw and geom_rangeframe instead of theme_jasp in variance component graph.
- Fixes https://github.com/jasp-stats/jasp-issues/issues/2274: Fixed calculation of y-axis limits of percent process variation graph in gauge linearity. It sometimes did not include zero in the limits and then removed all bars.
- Updated unit tests as graphs changed slightly after using jasp_theme_raw and geom_rangeframe instead of theme_jasp in bias and linearty graph and percent process variation graph.
- Fixes https://github.com/jasp-stats/jasp-issues/issues/2286: Tried to compare different data types and only aligned them when numeric. Now always aligning data type.

Based on https://github.com/jasp-stats/jaspQualityControl/pull/279 to make the unit tests work.

### CODE REVIEW

**Comment by JTPetter** [comment]
Probably best to review and merge this PR after https://github.com/jasp-stats/jaspQualityControl/pull/279, will rebase as soon as it is merged

**Comment by JTPetter** [comment]
Fixed another R4.3.1 issue that broke the reshaping from long to wide using a subgroup variable in process capability. From my side, this is ready to be reviewed now @vandenman

**Comment by JTPetter** [comment]
@vandenman I'm rebasing this now, could you try to review this before end of the week? This should definitely be in the upcoming release...

File: R/msaAttribute.R
Comment (vandenman) [inline R/msaAttribute.R:94]: This doesn't look quite right, but I guess this was already the case?
As in, two lines above you do
```r
jaspResults[["AAAtableGraphs"]] <- createJaspContainer(gettext("Attributes Agreement Analysis"))
```
and here you do
```r
jaspResults[["AAAtableGraphs"]] <- .aaaTableGraphs(...)
```
which means that you overwrite everything that was in `jaspResults[["AAAtableGraphs"]]` before. I mean, it probably works because it already worked before, but it's not very pretty.

**Review by vandenman — APPROVED** [review]
Two minor comments, if you have time for them they would be nice to fix but otherwise, this can be merged.

File: R/msaAttribute.R
Comment (JTPetter) [reply R/msaAttribute.R:94]: Yes, I think there is quite a bit in the measurement systems analysis that is not pretty, I would keep this for now, as I have a bigger rewrite of this part planned for early next year...

**Pull request merged**
